mysql-compatibility.md

```diff
@@ -137,7 +137,7 @@ As shown, because of the shared allocator, the `id` increments by 2 each time. T
 
 <CustomContent platform="tidb">
 
-TiDB utilizes a combination of [Prometheus and Grafana](/tidb-monitoring-api.md) for storing and querying performance monitoring metrics. In TiDB, performance schema tables do not return any results.
+TiDB utilizes a combination of [Prometheus and Grafana](/tidb-monitoring-api.md) for storing and querying performance monitoring metrics. In TiDB, most [performance schema tables](/performance-schema/performance-schema.md) do not return any results.
```

Comment:
Does this update also apply to the custom content for TiDB Cloud?

Reply:
Yes.

```
mysql-5.7.28-TiDB-Serverless> TABLE performance_schema.session_variables LIMIT 5;
+-----------------------------------+----------------+
| VARIABLE_NAME                     | VARIABLE_VALUE |
+-----------------------------------+----------------+
| innodb_io_capacity_max            | 2000           |
| tidb_non_prepared_plan_cache_size | 100            |
| version_compile_os                | linux          |
| net_retry_count                   | 10             |
| tidb_current_ts                   | 0              |
+-----------------------------------+----------------+
5 rows in set (0.04 sec)

mysql-5.7.28-TiDB-Serverless> SELECT TIDB_VERSION()\G
*************************** 1. row ***************************
TIDB_VERSION(): Release Version: v7.1.3-serverless
Edition: Community
Git Commit Hash: 2e8be50b073c26b8979c6988a3888ce57b5cec99
Git Branch: release-7.1-serverless
UTC Build Time: 2024-11-22 15:30:17
GoVersion: go1.21.11
Race Enabled: false
TiKV Min Version: 6.1.0
Check Table Before Drop: false
Store: tikv
1 row in set (0.02 sec)
```

So `performance_schema.session_variables` is not empty.

`performance_schema.session_connect_attrs` doesn't exist on TiDB Cloud Serverless, which is because v7.1.3 doesn't yet have this table. TiDB v7.5 does have this table.

Reply:
Looks like `session_variables` was introduced in v6.2.0

Reply:
On TiDB Cloud Serverless this is the only table with data.
```
$ mysqldump -u '4BFMphWRTf5tr4V.root' -h gateway01.eu-central-1.prod.aws.tidbcloud.com -P 4000 -pXXXXXXXXXX --skip-lock-tables --no-create-info performance_schema | awk '/^INSERT/ { print substr($0,0,100) }'
mysqldump: [Warning] Using a password on the command line interface can be insecure.
-- Warning: column statistics not supported by the server.
INSERT INTO `session_variables` VALUES ('tidb_build_stats_concurrency','4'),('tidb_opt_skew_distinct
```

On TiDB Cloud Dedicated there are most likely more.